test/Microsoft.AspNetCore.Server.KestrelTests/TestHelpers/MockSocket.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Threading;
+using Microsoft.AspNetCore.Server.Kestrel.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Networking;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests.TestHelpers
+{
+    class MockSocket : UvStreamHandle
+    {
+        public MockSocket(int threadId, IKestrelTrace logger) : base(logger)
+        {
+            var uv = new MockLibuv();
+            CreateMemory(uv, threadId, uv.handle_size(Libuv.HandleType.STREAM));
```

Comment:
This happens:

    System.NullReferenceException: Object reference not set to an instance of an object.
       at Microsoft.AspNetCore.Server.Kestrel.Networking.Libuv.unsafe_async_send(IntPtr handle) in D:\src\aspnet\KestrelHttpServer\src\Microsoft.AspNetCore.Server.Kestrel\Networking\Libuv.cs:line 226
       at Microsoft.AspNetCore.Server.Kestrel.Networking.UvAsyncHandle.ReleaseHandle() in D:\src\aspnet\KestrelHttpServer\src\Microsoft.AspNetCore.Server.Kestrel\Networking\UvAsyncHandle.cs:line 62
       at System.Runtime.InteropServices.SafeHandle.InternalFinalize()
       at System.Runtime.InteropServices.SafeHandle.Finalize()